components/nexus-commands/src/main/java/org/sonatype/nexus/commands/internal/FunctionMediator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.commands.internal;
+
+import java.util.Dictionary;
+
+import javax.inject.Named;
+
+import org.sonatype.nexus.commands.CommandHelper;
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import org.apache.felix.service.command.Function;
+import org.apache.karaf.shell.commands.Command;
+import org.eclipse.sisu.BeanEntry;
+import org.eclipse.sisu.Mediator;
+import org.osgi.framework.BundleContext;
+
+// TODO: Probably only need to support Action, but for now for testing registration of Functions too
```

Comment:
Functions don't participate fully with the Karaf shell, so I'd remove this rather than maintain it going forwards.